source/utilities.tex

```diff
@@ -21135,3 +21135,685 @@
+  template<> struct hash<stacktrace_entry>;
+  template<class Allocator> struct hash<basic_stacktrace<Allocator>>;
+}
+\end{codeblock}
+
+\rSec2[stacktrace.entry]{Class \tcode{stacktrace_entry}}
+
+\rSec3[stacktrace.entry.overview]{Overview}
+
+\begin{codeblock}
+namespace std {
+  class stacktrace_entry {
+  public:
+    using native_handle_type = @\impldef{\tcode{stacktrace_entry::native_handle_type}}@;
+
+    // \ref{stacktrace.entry.ctor}, constructors
+    constexpr stacktrace_entry() noexcept;
+    constexpr stacktrace_entry(const stacktrace_entry& other) noexcept;
+    constexpr stacktrace_entry& operator=(const stacktrace_entry& other) noexcept;
+
+    ~stacktrace_entry();
+
+    // \ref{stacktrace.entry.obs}, observers
+    constexpr native_handle_type native_handle() const noexcept;
+    constexpr explicit operator bool() const noexcept;
+
+    // \ref{stacktrace.entry.query}, query
+    string description() const;
+    string source_file() const;
+    uint_least32_t source_line() const;
+
+    // \ref{stacktrace.entry.cmp}, comparison
+    friend constexpr bool operator==(const stacktrace_entry& x,
+                                     const stacktrace_entry& y) noexcept;
+    friend constexpr strong_ordering operator<=>(const stacktrace_entry& x,
+                                                 const stacktrace_entry& y) noexcept;
+  };
+}
+\end{codeblock}
+
+\pnum
+An object of type \tcode{stacktrace_entry} is either empty,
+or represents a stacktrace entry and
+provides operations for querying information about it.
+The class \tcode{stacktrace_entry} models
+\tcode{regular}\iref{concepts.object} and
+\tcode{three_way_comparable<strong_ordering>}\iref{cmp.concept}.
+
+\rSec3[stacktrace.entry.ctor]{Constructor}
+
+\begin{itemdecl}
+constexpr stacktrace_entry() noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\ensures
+\tcode{*this} is empty.
+\end{itemdescr}
+
+\rSec3[stacktrace.entry.obs]{Observers}
+
+\begin{itemdecl}
+constexpr native_handle_type native_handle() const noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+The semantics of this function are
+\impldef{semantics of \tcode{stacktrace_entry::native_handle}}.
+
+\pnum
+\remarks
+Successive invocations of the \tcode{native_handle} function
+for an unchanged \tcode{stacktrace_entry} object return identical values.
+\end{itemdescr}
+
+\begin{itemdecl}
+constexpr explicit operator bool() const noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+\tcode{false} if and only if \tcode{*this} is empty.
+\end{itemdescr}
+
+\rSec3[stacktrace.entry.query]{Query}
+
+\pnum
+\begin{note}
+All the \tcode{stacktrace_entry} query functions treat
+errors other than memory allocation errors
+as ``no information available'' and do not throw in that case.
+\end{note}
+
+\begin{itemdecl}
+string description() const;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+A description of the evaluation represented by \tcode{*this},
+or an empty string.
```

Comment:
Shouldn't this read "and an empty string otherwise"? That is, when `*this` is empty or it an stacktrace entry not accurately represented.

Reply:
Where is the "if" part of the otherwise?

Reply:
Here: https://github.com/cplusplus/draft/pull/4375/files#diff-afa9f66c3857a24dc4092685af2cbf9cbc44bbd35afa4b0eb26a14a324b0999dR21149-R21158
And here: https://github.com/cplusplus/draft/pull/4375/files#diff-afa9f66c3857a24dc4092685af2cbf9cbc44bbd35afa4b0eb26a14a324b0999dR21234-R21240

Reply:
I don't understand.

The normative text, as it stands, allows two outcomes of the function, regardless of whether the *this is empty or not: An empty string, or the description of an evaluation. (The result can be an empty string even if the *this is not empty.)

Changing this would be a normative change that needs an LWG issue.

Reply:
> The result can be an empty string even if the *this is not empty.

Yes, that was the intent. That intent was discussed and approved in LEWG and LWG.